### ISSUE TITLE
UI Component Types

### ISSUE THREAD

**Issue by HeroicEric** [prototypal-io/untitled-ui#40]
## TL;DR;

Related to https://github.com/prototypal-io/untitled-ui/issues/33

I'm looking for confirmation that I understand the responsibilities of each of
the two types of components we're creating and that I use names that are the
least confusing to everyone else.

For the purposes of this discussion, let's try to refer to the two types of
components that we're currently using as "Outer" and "Kind" just so we don't
confuse each other :-p
## Outer UI Components
- Determines which "kind" component to render (and renders it)
- Provides its public API to the "kind" component (properties & actions)
- Has no presentational HTML or CSS

The outer UI component's main job is to provide behavior to and render the
correct "kind" component.
#### Example

`{{ui-panel}}`
#### Possible Names
- `UIComponent`
- `UIBaseComponent`
- `UIFrameComponent`
- `UIWrapperComponent`
- `UIOuterComponent`
- `UIBehaviorComponent`
## Kind Components
- Presentational markup & styles
- Take advantage of public API exposed by its "Outer" component
- Yield any related contextual components
- Extend behavior provided by its "Outer" component?
#### Example

`{{ui-panel--default}}`
#### Possible Names
- `UIKindComponent`
- `UIPresentationComponent`
## My Preference

I think `UIComponent` and `UIComponentKind` make sense so I'm going to go with that for now so I can work on implementing these and we can rename if needed.

**Comment by jacojoubert** [comment]
The idea behind the `outer` component was to provide all the functionality and primitives that would make building your own variation (`kind`) of the same thing really easy. Typically when you import a 3rd party component you get a bunch of css and then when you want to make it fit your app you have to fight with it to override css. For functional changes are are at the mercy of the component maintainer. 

The idea was that with `kind` components it would be really easy to build your own version of a  component and make it work exactly as you want, without having to rewrite a lot of the logic.

So a `outer` component can have css. The `button`, for example, has a corresponding `ui-button.scss` file that contains the css that makes the active state detection work. It is the only component right now that have this file.

What would the name structure be used for? Documentation? I would say `UIComponent` and `UIComponentKind` makes most sense, though `UIComponentKind` wasn't listed under the possible names.

**Comment by HeroicEric** [comment]
#50
